### ISSUE TITLE
test_builds_old_style_initless_class fails on Python 2.7 with typing backport installed

### ISSUE THREAD

**Issue by felixonmars** [HypothesisWorks/hypothesis#1793]
The test failure is new in hypothesis 4.5.0.

```
platform linux2 -- Python 2.7.15, pytest-4.2.0, py-1.7.0, pluggy-0.8.1

=================================== FAILURES ===================================
_____________________ test_builds_old_style_initless_class _____________________
Traceback (most recent call last):
  File "/build/python-hypothesis/src/hypothesis-4.5.0-py2/hypothesis-python/tests/py2/test_from_type.py", line 39, in test_builds_old_style_initless_class
    st.builds(OldStyleInitlessClass).example()
  File "/build/python-hypothesis/src/hypothesis-4.5.0-py2/hypothesis-python/src/hypothesis/_strategies.py", line 1147, in builds
    @defines_strategy
  File "/build/python-hypothesis/src/hypothesis-4.5.0-py2/hypothesis-python/src/hypothesis/_strategies.py", line 196, in cached_strategy
    if not isinstance(result, SearchStrategy) or result.is_cacheable:
  File "/build/python-hypothesis/src/hypothesis-4.5.0-py2/hypothesis-python/src/hypothesis/searchstrategy/strategies.py", line 153, in accept
    recur(self)
  File "/build/python-hypothesis/src/hypothesis-4.5.0-py2/hypothesis-python/src/hypothesis/searchstrategy/strategies.py", line 150, in recur
    mapping[strat] = getattr(strat, calculation)(recur)
  File "/build/python-hypothesis/src/hypothesis-4.5.0-py2/hypothesis-python/src/hypothesis/searchstrategy/lazy.py", line 100, in calc_is_cacheable
    if isinstance(v, SearchStrategy) and not v.is_cacheable:
  File "/usr/lib/python2.7/site-packages/typing.py", line 1259, in __instancecheck__
    return issubclass(instance.__class__, self)
AttributeError: class OldStyleInitlessClass has no attribute '__class__'
===Flaky Test Report===


===End Flaky Test Report===
========================== slowest 20 test durations ===========================
```

**Comment by Zac-HD** [comment]
It looks to me like that error is actually coming from the typing backport. Does it still happen with typing uninstalled? 

I'd also be surprised if it first appeared in 4.5.0, that's #1780 and doesn't seem related.

**Comment by felixonmars** [comment]
You are right, `typing` was brought in by new flake8 (3.7+) here.

**Comment by Zac-HD** [comment]
OK, we'll probably deal with this tomorrow when we do our weekly bump-the-dependencies PR! Otherwise I'll see if we can just skip that test when typing is installed.

**Comment by felixonmars** [comment]
Looks like typing was already pinned to 3.6.6 several months ago, so it wasn't caught by the weekly PR. I'm curious why this setup works in the CI though :/

BTW my patch for typing has been merged, but no new release so far.

**Comment by Zac-HD** [comment]
`requirements/typing.txt` has a pin to [3.6.6, which is the latest version on PyPI](https://pypi.org/project/typing/) - our weekly PR should update it from the unpinned one in `requirements/typing.in` when your patch is released.

It didn't happen this week because the flake8 job just hung forever, so I skipped that part of the upgrade - that kind of bug will often be fixed by next week, so I'll debug then if it isn't.

**Comment by Zac-HD** [comment]
I don't think there's anything we actually need to do here, so I'll call this issue closed by python/typing#612.
